script/import_scripts/yammer.rb

```diff
@@ -0,0 +1,505 @@
+# frozen_string_literal: true
+
+# Yammer importer
+#  https://docs.microsoft.com/en-us/yammer/manage-security-and-compliance/export-yammer-enterprise-data#export-yammer-network-data-by-date-range-and-network
+#
+# You will need a bunch of CSV files:
+#
+# - Users.csv Groups.csv Topics.csv Groups.csv Files.csv Messages.csv
+# (Others included in Yammer export are ignored)
+
+require "csv"
+require_relative "base"
+require_relative "base/generic_database"
+
+# Call it like this:
+#   RAILS_ENV=production bundle exec ruby script/import_scripts/yammer.rb DIRNAME
+
+class ImportScripts::Yammer < ImportScripts::Base
+  BATCH_SIZE = 1000
+  NUM_WORDS_IN_TITLE = ENV["NUM_WORDS_IN_TITLE"].to_i || 20
+  SKIP_EMPTY_EMAIL = true
+  SKIP_INACTIVE_USERS = false
+  PARENT_CATEGORY_NAME = ENV["PARENT_CATEGORY_NAME"] || "Yammer Import"
+  IMPORT_GROUPS_AS_TAGS = true
+  MERGE_USERS = true
+  # import groups as tags rather than as categories
+  SiteSetting.tagging_enabled = true if IMPORT_GROUPS_AS_TAGS
+  PM_TAG = ENV["PM_TAG"] || "eht"
+
+  def initialize(path)
+    super()
+
+    @path = path
+    @db = ImportScripts::GenericDatabase.new(@path, batch_size: BATCH_SIZE, recreate: true)
+  end
+
+  def execute
+    create_developer_users
+    read_csv_files
+
+    import_categories
+    import_users
+    import_topics
+    import_pm_topics
+    import_posts
+    import_pm_posts
+  end
+
+  def create_developer_users
+    GlobalSetting
+      .developer_emails
+      .split(",")
+      .each { |e| User.create(email: e, active: true, username: e.split("@")[0]) }
+  end
+
+  def read_csv_files
+    puts "", "reading CSV files"
+
+    # consider csv_parse Tags.csv
+    # consider Admins.csv that has admins
+
+    u_count = 0
+    csv_parse("Users") do |row|
+      next if SKIP_INACTIVE_USERS && row[:state] != "active"
+      u_count += 1
+      @db.insert_user(
+        id: row[:id],
+        email: row[:email],
+        name: row[:name],
+        username: row[:name],
+        bio: "",
+        # job_title: row[:job_title],
```

Comment:
Probably because it's a copy of a different import script? :shrug:
I didn't want to invest too much time cleaning up the script. I assume we will need to give it some more love when/if we actually get to use the script for a customer. Until then, it seemed good enough for a start.